src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,143 @@
 package core.basesyntax;
 
-public class MyHashMap<K, V> implements MyMap<K, V> {
+import java.util.Map;
+import java.util.Objects;
 
-    @Override
-    public void put(K key, V value) {
+public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
 
+    static int hash(Object key) {
+        int keyHash;
+        return (key == null) ? 0 : (keyHash = key.hashCode()) ^ (keyHash >>> 16);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        if (table != null && table.length > 0) {
+            int index = (table.length - 1) & hash;
+            if (table[index] != null) {
+                Node<K, V> currentNode = table[index];
+                while (currentNode != null) {
+                    if (currentNode.hash == hash
+                            && currentNode.key == key
+                            || key != null && key.equals(currentNode.key)) {
```

Comment:
In the `getValue` method, the key comparison should use `Objects.equals(currentNode.key, key)` instead of `==` to ensure proper comparison of keys, especially for non-interned strings or objects.